cmd/clusterctl/cmd/generate_provider.go

```diff
@@ -89,6 +90,8 @@ func init() {
 		"IPAM provider and version (e.g. infoblox:v0.0.1)")
 	generateProviderCmd.Flags().StringVar(&gpo.runtimeExtensionProvider, "runtime-extension", "",
 		"Runtime extension provider and version (e.g. test:v0.0.1)")
+	generateProviderCmd.Flags().StringVar(&gpo.addonProvider, "add-on", "",
```

Comment:
```suggestion
	generateProviderCmd.Flags().StringVar(&gpo.addonProvider, "addon", "",
```
To match flag defined in delete (same in other commands)